### ISSUE TITLE
morphology: the finite max_val=1e4 sentinel leaks into the output (empty windows, |x| >= 1e4) and caps the convolution engine at one ULP of max_val

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4734]
## Summary

Every `kornia.morphology` function stands in for the infinities of the structuring element with a finite `max_val = 1e4`: outside pixels are padded with `-max_val` (dilation) / `+max_val` (erosion) and masked-out kernel cells contribute `f - max_val`. The sentinel is an ordinary float that reaches the output whenever (a) no kernel member lands inside the image, or (b) the image values approach `max_val` in magnitude. It also caps the accuracy of the `convolution` engine at one ULP of `max_val`.

```python
import torch
from kornia.morphology import dilation, erosion

# (a) an empty window returns the sentinel instead of ignoring the pixel
dilation(torch.ones(1, 1, 1, 1), torch.tensor([[1., 0., 0.]]), origin=[0, 0]).item()   # -9999.0  (skimage mode="ignore": 1.0; scipy: -inf)
erosion(torch.ones(1, 1, 1, 1), torch.tensor([[1., 0., 0.]]), origin=[0, 2]).item()    # 10000.0

# (b) values comparable to max_val: a masked-out cell wins with f - 1e4
dilation(torch.tensor([[0., 5e4, 0.]])[None, None], torch.tensor([[1., 0., 1.]]))    # [50000., 40000., 50000.]  (true: [50000., 0., 50000.]; correct with max_val=1e6)
erosion(torch.tensor([[0., -5e4, 0.]])[None, None], torch.tensor([[1., 0., 1.]]))    # [-50000., -40000., -50000.]
dilation(torch.tensor([[-5e4, -6e4]])[None, None], torch.ones(1, 3))                  # [-10000., -10000.]  the border pad wins

# (c) the convolution engine mixes the sentinel through conv2d and loses one ULP of max_val
x = torch.rand(1, 1, 9, 11, generator=torch.Generator().manual_seed(0)); k = torch.ones(3, 3)
for mv in (1.0, 1e2, 1e4, 5e4):
    d = (dilation(x, k, max_val=mv, engine="unfold") - dilation(x, k, max_val=mv, engine="convolution")).abs().max().item()
    print(mv, d)   # 1.19e-07, 7.15e-06, 1.27e-03, 4.78e-03  — exactly ULP(max_val); 0 in float64
```

Measured on `main` @ 0e8cf3b57, torch 2.14.0, CPU float32 (macOS arm64; the conv gap survives `torch.backends.mkldnn.flags(enabled=False)` and vanishes in float64, so it is the sentinel magnitude, not a platform quirk). uint16-range data, depth in millimetres, HDR and integer label maps all sit outside `(-1e4, 1e4)`. The `unfold` engine itself is exact in float16/bfloat16 below `max_val`.

## Expected

Outside pixels and masked cells are excluded, not given a large finite value: pad with `-inf`/`+inf` (or `torch.finfo(dtype).max`) and mask instead of adding, so that an empty window returns the identity of the operation (as skimage `mode="ignore"`) or a documented value, and the result does not depend on the image range. The `max_val` argument then becomes unnecessary. Until then the docstring must say that `max_val` is a finite sentinel that has to exceed the image's absolute range and that it bounds the `convolution` engine's accuracy (batch 7 documents it; this issue is the fix).

## Classification

Bug (clearly broken output outside `(-max_val, max_val)` and for empty windows). Idiom: candidate, not numbered: *a finite sentinel standing in for +-inf inside the input's value range* (sibling of idiom 1's `x + eps`; same family as the uint8 `max() + 1` wrap of #4695).

## Disposition

`n/a — single symbol` (all seven functions share the two implementations). Focused fix, welcome as a PR.

Related: #4713, #4525 / #4729 / #4728 (engine work that would inherit the same sentinel).

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).

**Comment by NiteeshGowda72** [comment]
i am working on it
